src/conference_scheduler/scheduler.py

```diff
@@ -74,18 +76,23 @@ def heuristic(events,
 
         [(0, 1), (1, 4), (2, 5)]
     """
-    X = heu.get_initial_array(events=events, slots=slots)
-
     def count_violations(array):
         return len(list(val.array_violations(array, events, slots)))
 
-    X = algorithm(initial_array=X,
-                  objective_function=count_violations,
-                  lower_bound=0,
-                  **initial_solution_algorithm_kwargs)
+    if initial_solution is None:
+        X = heu.get_initial_array(events=events, slots=slots)
+        X = algorithm(initial_array=X,
+                      objective_function=count_violations,
+                      lower_bound=0,
+                      **initial_solution_algorithm_kwargs)
+    else:
+        X = initial_solution
 
     if objective_function is not None:
 
+        if objective_function is of.equity_capacity_demand_difference:
```

Comment:
I can do without this check @meatballs, so just have that `"beta"` is passed to all objective functions whether or not it is needed. I think that might be nicer: any preference?

Reply:
I reckon without is nicer

Reply:
sorry, without the condition - pass the beta whether it's needed or not

Reply:
Yup! Doing that now.